fastcrypto/src/groups/mod.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// Impl operator overload for the elements of a given group.
+/// * `scalar` -  Type of scalars. Should be equal to group::Scalar.
+/// * `element` - Type of elements. Should be equal to group::Element.
+/// * `group` - Type of group.
+#[macro_export]
+macro_rules! impl_group {
```

Comment:
This macro ensures that operator overloads are implemented for all combinations of borrowed and owned instances.